File: mutilMoudle_scalaMoudle/src/main/scala/com/test/mutilMoudle_scalaMoudle/class_object/Person.scala
package com.test.mutilMoudle_scalaMoudle.class_object

/**
  * Person类
  * @param name
  * @param age
  */
//主构造方法
class Person(val name:String, var age:Int) {
  var score:Double = _
  var grade:String = _

  //辅助构造方法，辅助构造方法，首先要调用主构造方法或者其他构造方法
  def this(name:String ,age:Int, score:Double){
    this(name, age)
    this.score = score
  }

  def this(name:String, age:Int, score:Double, grade:String){
    this(name, age, score)
    this.grade = grade
  }

  override def toString = s"Person($score, $grade, $name, $age)"
}

/**
  * Person类的伴生对象
  */
object Person{

  def apply(name:String, age:Int): Person ={
    new Person(name, age)
  }

  def apply(name:String, age:Int, score:Double): Person ={
    new Person(name, age,score)
  }

  def apply(name:String, age:Int, score:Double, grade:String): Person ={
    new Person(name, age, score, grade)
  }

}

File: mutilMoudle_javaMoudle/src/main/java/com/test/mutilMoudle_javaMoudle/Person.java
package com.test.mutilMoudle_javaMoudle;

/**
 *Auther:      吕学文<lvxw@fun.tv>
 *Date:        20180205
 *Description: 这是一个实体类：
 *                  1、用于封装前端传递的参数
 *                  2、数据库记录映射
 */
public class Person {
    private String name;
    private int age;
    private double score;

    /**
     *Auther:      吕学文<lvxw@fun.tv>
     *Date:        20180205
     *该方法用于计算整的年龄
     * @param deviation 年龄的偏移量。负数表示,年龄向下取整数；正数表示年龄向上取整
     * @return  返回的是取整后的年龄
     */
    public int getNeatAge(int deviation){
        //。。。。。。（假如这块逻辑比较复杂）
        int neatAge = this.getAge()+deviation;
        return neatAge;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public double getScore() {
        return score;
    }

    public void setScore(double score) {
        this.score = score;
    }

    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", age=" + age +
                ", score=" + score +
                '}';
    }
}
